banks-server/src/banks_server.rs

```diff
@@ -180,6 +186,13 @@ impl Banks for BanksServer {
             slot,
             confirmations,
             err: status.err(),
+            confirmation_status: if confirmations.is_none() {
+                Some(TransactionConfirmationStatus::Max)
```

Comment:
It doesn't look like the bank server logic matches rpc server logic so I don't think we can set this as `Max` quite yet. We don't have this check: https://github.com/solana-labs/solana/blob/d2b0fd973f91bb0c57f51e7762dfa592344e158a/core/src/rpc.rs#L889

Reply:
Looks like we could update BanksServer to use Max, since it has access to the BlockCommitmentCache. The current behavior already has this issue (not matching RpcClient), since it sets confirmations to None when a transaction is rooted.

My main intent was to keep the BanksServer and RpcClient interfaces the same, although I suppose that isn't strictly necessary. We could do with BanksServer whatever best suits ProgramTest, since that is the only downstream user.

Reply:
Yup, keeping the interfaces the same sounds great, just wanted to call out the behavior difference. Could be a follow up fix

Reply:
Perfect, I will update BanksServer in a follow-up.